org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/scan/DeviceRequest.java

```diff
@@ -318,6 +360,38 @@ public void setDeviceType(DeviceType deviceType) {
 		this.deviceType = deviceType;
 	}
 
+	public String getAttributeName() {
+		return attributeName;
+	}
+
+	public void setAttributeName(String attributeName) {
+		this.attributeName = attributeName;
+	}
+
+	public boolean isGetAllAttributes() {
+		return getAllAttributes;
+	}
+
+	public void setGetAllAttributes(boolean getAllAttributes) {
+		this.getAllAttributes = getAllAttributes;
+	}
+
+	public Map<String, IDeviceAttribute<?>> getAttributes() {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)